### ISSUE TITLE
Code Error?

### ISSUE THREAD

**Issue by 040840308** [carlini/audio_adversarial_examples#17]
Hi
      I wonder whether this is the error in the code (line 239). 
      # Actually do the optimization ste
            d, el, cl, l, logits, new_input, _ = sess.run((self.delta, self.expanded_loss,
                                                           self.ctcloss, self.loss,
                                                           self.logits, self.new_input,
                                                           self.train),
                                                          feed_dict)
      Should the self.delta be the self.apply_delta?
      
      Thanks

**Comment by carlini** [comment]
I don't think so. `self.apply_delta` is the clipped value, and I want to look at the un-clipped value `self.delta`. What makes you think it should be the other way around?

**Comment by 040840308** [comment]
Since the self.delta is initialized as value zeros:
self.delta = delta = tf.Variable(np.zeros((batch_size, max_audio_len), dtype=np.float32), name='qq_delta')

In the following code, the self.delta is never updated except:
if finetune is not None and len(finetune) > 0:
            sess.run(self.delta.assign(finetune-audio))
However, if the default fineture is None, the self.delta will be always zeros. Is this right or my misunderstanding about this?

**Comment by carlini** [comment]
Ah, it's updated here:
```
        grad,var = optimizer.compute_gradients(self.loss, [delta])[0]
        self.train = optimizer.apply_gradients([(tf.sign(grad),var)])
```

**Comment by 040840308** [comment]
OK, I see. Maybe, I still have some misunderstanding about tensorflow. Thanks

**Comment by carlini** [comment]
Alright. Closing this for now then.
